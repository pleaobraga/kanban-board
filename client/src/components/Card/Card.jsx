import './Card.scss'

import React from 'react'
import PropTypes from 'prop-types'
import { startCase, kebabCase } from 'lodash'
import { Draggable } from 'react-beautiful-dnd'
import { useDispatch } from 'react-redux'

import { deleteCard } from '../../reducers/BoardReducer'

export const Card = ({
  id,
  index,
  type,
  duration,
  severity,
  TaskListIndex
}) => {
  const dispatch = useDispatch()

  const deleteItem = () => {
    //TODO Implemen delete function

    const card = {
      id,
      index,
      TaskListIndex
    }

    dispatch(deleteCard(card))
  }

  return (
    <Draggable draggableId={id} index={index}>
      {(provided, snapshot) => (
        <div
          className={`card card--${kebabCase(type)} ${snapshot.isDragging &&
            'card--dragging'}`}
          ref={provided.innerRef}
          {...provided.draggableProps}
          {...provided.dragHandleProps}
        >
          <i className="material-icons close-icon" onClick={deleteItem}>
            close
          </i>
          <h3 className="card__type">{startCase(type)}</h3>
          <h4 className="card__duration">{`${duration} hr${
            duration > 1 ? 's' : ''
          }`}</h4>
          <h3 className="card__severity">{startCase(severity)}</h3>
        </div>
      )}
    </Draggable>
  )
}

Card.propTypes = {
  id: PropTypes.string.isRequired,
  index: PropTypes.number.isRequired,
  type: PropTypes.oneOf(['feature', 'bugFix', 'update', 'research', 'content'])
    .isRequired,
  duration: PropTypes.number.isRequired,
  severity: PropTypes.oneOf(['hight', 'medium', 'low']).isRequired,
  TaskListIndex: PropTypes.number.isRequired
}

export default React.memo(Card)
