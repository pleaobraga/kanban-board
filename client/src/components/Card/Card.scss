@import '../../styles/_color.scss';

.card {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: space-around;

  clip-path: polygon(50% 0%, 100% 0, 100% 88%, 81% 100%, 0 100%, 0 0);

  position: relative;

  width: 21rem;
  height: 15rem;

  padding: 0 2.2rem;

  &:hover .close-icon {
    display: block;
  }

  .close-icon {
    position: absolute;
    top: 5px;
    right: 6px;
    display: none;

    font-size: 3rem;

    width: 3rem;
    height: 3rem;

    cursor: pointer;
  }

  &:before {
    content: '';

    position: absolute;
    bottom: 0;
    right: 0;
    display: block;

    width: 4rem;
    height: 2rem;

    background: black;
    opacity: 0.2;
  }

  &--feature {
    background: $blue-light;
  }

  &--update {
    background-color: $orange;
  }

  &--bug-fix {
    background-color: $salmon;
  }

  &--research {
    background-color: $purple;
  }

  &--content {
    background-color: $green;
  }

  &--dragging {
    background-color: lightgreen;
  }

  &__type {
    font-family: 'Handlee', cursive;
    font-size: 4rem;
    color: $dark-blue;
    text-align: left;
  }

  &__duration {
    font-family: 'Handlee', cursive;
    font-size: 3rem;
    color: $white;
    text-align: left;
  }

  &__severity {
    font-family: 'Handlee', cursive;
    font-size: 3rem;
    color: $white;
    text-align: cenleftter;
  }
}
