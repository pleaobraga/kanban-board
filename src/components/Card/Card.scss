@import '../../styles/_colors.scss';

.card {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-around;

  clip-path: polygon(50% 0%, 100% 0, 100% 88%, 81% 100%, 0 100%, 0 0);

  position: relative;

  width: 20rem;
  height: 15rem;

  &:before {
    content: '';

    position: absolute;
    bottom: 0;
    right: 0;
    display: block;

    width: 4rem;
    height: 2rem;

    background: black;
    opacity: 0.2;
  }

  &__feature {
    background: $blue;
  }

  &__update {
    background-color: $orange;
  }

  &__bug-fix {
    background-color: $salmon;
  }

  &__research {
    background-color: $salmon;
  }

  &__content {
    background-color: $green;
  }

  &--type {
    font-family: 'Handlee', cursive;
    font-size: 4rem;
    color: $dark-blue;
    text-align: left;
  }

  &--duration {
    font-family: 'Handlee', cursive;
    font-size: 3rem;
    color: $white;
    text-align: left;
  }

  &--severity {
    font-family: 'Handlee', cursive;
    font-size: 3rem;
    color: $white;
    text-align: cenleftter;
  }
}
