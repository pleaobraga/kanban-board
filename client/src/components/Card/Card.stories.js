import React from 'react'
import { storiesOf } from '@storybook/react'
import { number, select } from '@storybook/addon-knobs'
import { DragDropContext, Droppable } from 'react-beautiful-dnd'

import ProviderWrapper from '../../../.storybook/provider'
import store from '../../../.storybook/configureStore'
import { board } from '../../reducers/__mocks__/reduxMock'
import { Card } from './Card'

const initialState = {
  board,
  loadingContent: false,
  errorContent: false
}

const newStore = store(initialState)

const withProvider = story => (
  <ProviderWrapper store={newStore}>{story()}</ProviderWrapper>
)

const getContent = () => {}

export default storiesOf('Components | Card', module)
  .addDecorator(withProvider)
  .add(
    'default',
    () => (
      <DragDropContext>
        <Droppable droppableId="droppable-1" type="PERSON">
          {provided => (
            <div ref={provided.innerRef} {...provided.droppableProps}>
              <Card
                getContent={getContent}
                id={'0'}
                index={0}
                type={select(
                  'Type',
                  ['feature', 'bug fix', 'update', 'research', 'content'],
                  'feature'
                )}
                duration={number('Duration', 1)}
                severity={select(
                  'Severity',
                  ['hight', 'medium', 'low'],
                  'medium'
                )}
                TaskListIndex={1}
              />
            </div>
          )}
        </Droppable>
      </DragDropContext>
    ),
    {
      info: {
        inline: true,
        header: false,
        source: false,
        propTables: [Card]
      }
    }
  )
