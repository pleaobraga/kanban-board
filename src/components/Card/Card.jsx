import './Card.scss'

import React from 'react'
import PropTypes from 'prop-types'
import { startCase, kebabCase } from 'lodash'
import { Draggable } from 'react-beautiful-dnd'

export const Card = ({ id, index, type, duration, severity }) => {
  const deleteItem = () => {
    //TODO Implemen delete function
  }

  return (
    <Draggable draggableId={id} index={index}>
      {provided => (
        <div
          className={`card card__${kebabCase(type)}`}
          ref={provided.innerRef}
          {...provided.draggableProps}
          {...provided.dragHandleProps}
        >
          <i className="material-icons close-icon" onClick={deleteItem}>
            close
          </i>
          <h3 className="card--type">{startCase(type)}</h3>
          <h4 className="card--duration">{`${duration} hr${
            duration > 1 ? 's' : ''
          }`}</h4>
          <h3 className="card--severity">{startCase(severity)}</h3>
        </div>
      )}
    </Draggable>
  )
}

Card.propTypes = {
  id: PropTypes.string.isRequired,
  index: PropTypes.number.isRequired,
  type: PropTypes.oneOf(['feature', 'bugFix', 'update', 'research', 'content'])
    .isRequired,
  duration: PropTypes.number.isRequired,
  severity: PropTypes.oneOf(['hight', 'medium', 'low']).isRequired
}

export default React.memo(Card)
