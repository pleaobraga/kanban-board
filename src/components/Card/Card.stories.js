import React from 'react'
import { storiesOf } from '@storybook/react'
import { number, select } from '@storybook/addon-knobs'
import { Card } from './Card'

export default storiesOf('Components | Card', module).add(
  'default',
  () => (
    <Card
      type={select(
        'Type',
        ['feature', 'bug fix', 'update', 'research', 'content'],
        'feature'
      )}
      duration={number('Duration', 1)}
      severity={select('Severity', ['hight', 'medium', 'low'], 'medium')}
    />
  ),
  { info: { inline: true, header: false } }
)
