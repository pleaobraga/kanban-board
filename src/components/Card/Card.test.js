import React from 'react'
import { mount } from 'enzyme'
import Card from './Card'

describe('Card', () => {
  const card = mount(<Card title="Test" text="test" />)

  it('render properly', () => {
    expect(card).toMatchSnapshot()
  })
})
