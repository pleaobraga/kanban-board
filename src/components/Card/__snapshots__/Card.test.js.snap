// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card render properly 1`] = `
<PublicDraggable
  draggableId={0}
  index={0}
>
  <Component />
</PublicDraggable>
`;
